base_layer/wallet/src/output_manager_service/storage/database/backend.rs

```diff
@@ -37,29 +38,32 @@ pub trait OutputManagerBackend: Send + Sync + Clone {
     /// Modify the state the of the backend with a write operation
     fn write(&self, op: WriteOperation) -> Result<Option<DbValue>, OutputManagerStorageError>;
     fn fetch_pending_incoming_outputs(&self) -> Result<Vec<DbWalletOutput>, OutputManagerStorageError>;
-
-    fn set_received_output_mined_height_and_status(
+    /// Perform a batch update of the received outputs' mined height and status
+    fn set_received_outputs_mined_height_and_status_batch_mode(
```

Comment:
```suggestion
    fn set_received_outputs_mined_height_and_status(
```
NIT
I dont like the added `batch_mode` text to trait.
This leaks internal workings to the outer API. 
And the way you did in internally using a single or batched query, is the correct implementation that most people would assume its implemented if you read that the function takes in multiple fields.

Reply:
Yeah, the `_batch_mode` suffix is just to support these changes for now, but will disappear when the PR is ready for final review.